processing/src/main/java/io/druid/segment/data/CompressionFactory.java

```diff
@@ -0,0 +1,275 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.segment.data;
+
+import com.google.common.base.Supplier;
+import com.google.common.collect.Maps;
+import org.apache.commons.lang.ArrayUtils;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.util.Map;
+
+/**
+ * Compression of metrics is done by using a combination of {@link io.druid.segment.data.CompressedObjectStrategy.CompressionStrategy}
+ * and Encoding(such as {@link LongEncoding} for type Long). CompressionStrategy is unaware of the data type
+ * and is based on byte operations. It must compress and decompress in block of bytes. Encoding refers to compression
+ * method relies on data format, so a different Encoding exist for each data type.
+ * <p>
+ *
+ * Compression Storage Format
+ * Byte 1 : version (currently 0x02)
+ * Byte 2 - 5 : number of values
+ * Byte 6 - 9 : size per block (even if block format isn't used, this is needed for backward compatibility)
+ * Byte 10 : compression strategy (contains a flag if there's an encoding byte, see below for how the flag is defined)
+ * Byte 11(optional) : encoding type
+ * <p>
+ * Encoding specific header (described below)
+ * <p>
+ * Block related header (if block compression is used, described in GenericIndexed)
+ * <p>
+ * Values
+ */
+public abstract class CompressionFactory
+{
+
+  public static final LongEncoding DEFAULT_LONG_ENCODING = LongEncoding.LONGS;
+
+  /**
+   * Delta Encoding Header v1:
+   * Byte 1 : version
+   * Byte 2 - 9 : base value
+   * Byte 10 - 13 : number of bits per value
+   */
+  public static final byte DELTA_ENCODING_VERSION = 0x1;
+
+  /**
+   * Table Encoding Header v1 :
+   * Byte 1 : version
+   * Byte 2 - 5 : table size
+   * Byte 6 - (6 + 8 * table size - 1) : table of encoding, where the ith 8-byte value is encoded as i
+   */
+  public static final byte TABLE_ENCODING_VERSION = 0x1;
+
+  public static final int MAX_TABLE_SIZE = 256;
+
+  /**
```

Comment:
This is a javadoc style comment (`/**`) but it is not actually attached to anything. It should just be a regular multiline comment (`/*`).